veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,311 @@
+
+However, container runtimes drop all capabilities from the [bounding set](https://man7.org/linux/man-pages/man7/capabilities.7.html), and capabilities must explicitly be added in the SecurityContext, as done for CAP_NET_BIND_SERVICE in [virt-launcher](https://github.com/kubevirt/kubevirt/blob/release-1.5/pkg/virt-controller/services/rendercontainer.go#L284).
+
+Since virt-launcher drops the root user, this method is secure. However, KubeVirt follows Kubernetes' `Restricted` Pod Security Standard, which disallows CAP_NET_ADMIN.
+
+## Scalability
+
+- `passt` has a memory overhead of ~250Mi per VM. Users running VMs at scale should revert to other bindings if memory overhead is a concern.
+-  VMs with a large number of TCP connections will take longer to live migrate. That's because sockets are handled in a serial
+ fashion, each one separately sent back and forth to passt-repair for handling.
+- The number of concurrent live migration per node is already limited and bound by the number of virt-handler worker goroutines.
+
+## Update/Rollback Compatibility
+
+- Live migration of VMs with `passt` network binding, from nodes running older versions of KubeVirt to an upgraded KubeVirt environment will result in TCP connections being reset. 
+The migration will succeed, but since the old instance of passt does not have any of the necessary code and infrastructure to properly freeze the socket, record the state and send the state to the target `passt` instance, it will receive nothing, and not attempt to restore the socket state. From the client perspective, the connection would be reset and a new handshake will be necessary.      
+- Rollback of passt migration functionality (in case system stability is impacted) will be possible by disabling of the feature gate. It is expected that passt would still work, but live migrations would disconnect TCP connections.
+
+## Functional Testing
+
+Existing e2e tests cover basic passt connectivity and remain sufficient with vhost-user support. Dedicated e2e tests for seamless migration will verify TCP connection persistence. Sticky IP addresses are required, so tests will use secondary interfaces with hardcoded IPs.
+
+## Implementation Phases
+
+1. Implement vhost-user in passt-sidecar.
+2. Implement calls to passt-repair in virt-handler.
+3. Enhance e2e tests and demonstrate migration without disconnection.
+
+Phases can proceed in parallel.
+
+## Feature Lifecycle Phases
+
+`passt` is currently a network binding plugin external to the KubeVirt core, and its use as a network binding is not subject to lifecycle phases. However, the calls to passt-repair from virt-handler impose a risk and should be protected by a feature gate and subject to lifecycle phases.
+
+### Alpha
+
+`passt live migration` will be feature-gated, requiring explicit user opt-in. virt-handler will not run `passt-repair` unless the feature gate is enabled.
```

Comment:
How about `passtTCPMigration` ?

Reply:
That might not be the best name.  TCP is the tricky case and the one we've implemented so far.  There are some more unusual edge cases for UDP which would require some additional migration support, which we hope to implement eventually.  When we get to that we don't expect it to require changes on the KubeVirt side.

Reply:
Then how about `passtIPStackMigration` ?